dali/c_api/c_api.cc

```diff
@@ -406,15 +406,42 @@ void daliOutputCopy(daliPipelineHandle *pipe_handle, void *dst, int output_id,
   assert(ws != nullptr);
 
   auto &type_info = dali::TypeTable::GetTypeInfo(dali::DALIDataType::DALI_UINT8);
-  if (ws->OutputIsType<dali::CPUBackend>(output_id)) {
-    CopyToExternal(dst, dst_alloc_type, ws->OutputRef<dali::CPUBackend>(output_id),
+  if (ws->OutputIsType<dali::CPUBackend>(output_idx)) {
+    CopyToExternal(dst, dst_alloc_type, ws->OutputRef<dali::CPUBackend>(output_idx),
                    stream, sync, use_copy_kernel);
   } else {
-    CopyToExternal(dst, dst_alloc_type, ws->OutputRef<dali::GPUBackend>(output_id),
+    CopyToExternal(dst, dst_alloc_type, ws->OutputRef<dali::GPUBackend>(output_idx),
                    stream, sync, use_copy_kernel);
   }
 }
 
+void daliOutputCopySamples(daliPipelineHandle *pipe_handle, void **dsts, int output_idx,
+                           device_type_t dst_type, cudaStream_t stream, unsigned int flags) {
+  dali::TimeRange tr("daliOutputCopySamples", dali::TimeRange::kGreen);
+
+  bool is_pinned = flags & DALI_ext_pinned;
+  bool sync = flags & DALI_ext_force_sync;
+  bool use_copy_kernel = flags & DALI_use_copy_kernel;
+
+  using dali::kernels::AllocType;
+  AllocType dst_alloc_type = dst_type == device_type_t::GPU
+        ? AllocType::GPU
+        : (is_pinned ? AllocType::Pinned : AllocType::Host);
```

Comment:
Maybe you can wrap this into function as this pattern appears in more than one place.